vm-transfer/zvm-transfer.py

```diff
@@ -0,0 +1,455 @@
+                                       tags=filtered_tags,
+                                       boot_volumes=[{"id": vm.bootVolume, "disk_bus": "virtio", "device_type": "disk"}],
+                                       volumes_to_attach=vm.volumes,
+                                       hw_firmware_type=vm.hw_firmware_type,
+                                       networks=networks,
+                                       guest_os=guest_os,
+                                       os_type_id=vm.provided_os_type_id,
+                                       powerup=False)
+    logger.info("Created VM:\n%s", pformat(created_vm))
+
+
+def init_logger():
+    formatter = logging.Formatter('%(asctime)s [%(name)s] %(levelname)-10s %(message)s')
+    console_handler = logging.StreamHandler(sys.stdout)
+    console_handler.setFormatter(formatter)
+    logger.addHandler(console_handler)
+
+    if not os.path.exists(ZVM_CLONE_LOGS_DIR):
+        os.makedirs(ZVM_CLONE_LOGS_DIR)
+
+    logfile = '{logger_name}-{vm_id}.log'.format(logger_name=LOGGER_NAME, vm_id=args.vm_id)
+    logfile_with_path = os.path.join(ZVM_CLONE_LOGS_DIR, logfile)
+
+    file_handler = logging.FileHandler(filename=logfile_with_path)
+    atexit.register(file_handler.close)
+    file_handler.setFormatter(logging.Formatter('%(asctime)s %(levelname)-10s %(message)s \t'
+                                                '(%(pathname)s:%(lineno)d)'))
+    logger.addHandler(file_handler)
+
+    for handler in logger.handlers:
+        handler.set_level = logging.DEBUG
+    logger.setLevel(logging.DEBUG)
+
+    logger.info("Logger initialized")
+
+
+def get_to_ipdb():
+    src_client = init_src_symp_client()
+    dst_client = init_dst_symp_client()
+    import ipdb; ipdb.set_trace()
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("--vm-id", help="VM uuid", required=False)
+    parser.add_argument("--skip-sg", action='store_true', help="skip security-groups", default=False, required=False)
+    parser.add_argument("--ipdb", action='store_true', help="give me pdb with client", default=False, required=False)
+    parser.add_argument("--volume-id", help="Just manage volume", default=False, required=False)
+    parser.add_argument("--volume-name", help="Name for managed volume", default=None, required=False)
+
+    # Specify output of "--version"
+    parser.add_argument(
+        "--version",
+        action="version",
+        version="%(prog)s (version {version})".format(version=__version__))
+
+    return parser.parse_args()
+
+
+if __name__ == "__main__":
+    """ This is executed when run from the command line """
+    args = parse_arguments()
+    init_logger()
+    arguments = args
+    if args.vm_id:
+        migrate_vm(args.vm_id)
+        sys.exit(0)
+    if args.volume_id:
+        manage_single_volume(args.volume_id, args.volume_name)
+        sys.exit(0)
+    if args.ipdb:
+        get_to_ipdb()
+        sys.exit(0)
```

Comment:
This if statement (if args.ipdb) should probably be placed before the other if statements.